doc/src/explanation/solve_output.rst

```diff
@@ -0,0 +1,120 @@
+.. _solve_output:
+
+==========================
+Understanding Solve Output
+==========================
+
+The output of the `solve` function can seem very unwieldy since it may appear to
+arbitrarily return one of 6 different types of output (in addition to raising
+errors). The reasons for this are historical and are biased toward human
+interaction rather than programmatic use. The type of output will depend on the
+type of equation(s) (and how they are entered) and the number of symbols that
+are provided (and how they are provided).
+
+If consistent output from solve is desired use ``dict=True`` to get a list of
+solutions as dictionaries or ``set=True`` to get a tuple containing the list
+of variables and the set of values for those variables. (These settings only
+affect the output when the input does not contain relational expressions.)
+
+    >>> from sympy import sqrt, exp, solve, Symbol, Eq
+    >>> from sympy.abc import x, y, z, a, b
+    >>> solve(x, x, dict=True)
+    [{x: 0}]
+    >>> solve([x**2 - y, x + y - 6], set=True)
+    ([x, y], {(-3, 9), (2, 4)})
+
+The following is an explanation about the conditions under which various
+forms of output that are obtained by *default*.
+
+1. An empty list (or set) indicates that no solution was found.
+
+    >>> [] == solve(sqrt(x) + 1) == solve(sqrt(x) + 1, dict=True)
```

Comment:
This may just be personal preference: I find it easier to follow when the command is given and then the result, rather than giving the result along with the command. For example, I'd find this easier to understand:

```
>>> solve(sqrt(x) + 1)
[]
```

Granted, that would add a few lines to this docstring.